### ISSUE TITLE
report information when vscode and wingc crashes 

### ISSUE THREAD

**Issue by ekeren** [winglang/wing#589]
### Summary

when vscode / wingc panics we would like get notified with the right information

### Feature Spec

(I am using sentry as a placeholder, for the mechanism for crash reporting)

wing vscode plugins and wingc sends its' crash reports to https://sentry.io/organizations/winglang
- See instructions on how to reference a crash in a ticket 
- See instructions on how you can  opt out from sending crash reports in the plugin preference 
- See instructions on how you opt out `wingc` from sending crash report
- crash reports are open for our community, here are the instructions on how you can log into sentry winglang organization so you can view the current crash reports 

### Use Cases

The vscode is crushing and we do not know about it
wingc panics and we are not aware of it

### Implementation Notes

We could consider using some saas solution for crash reports (e.g. sentry) if we provide the ability to opt out from reporting

### Component

Compiler, IDE Extension

**Comment by staycoolcall911** [comment]
Keep

**Comment by hasanaburayyan** [comment]
I have used sentry in a few places before and really like it. I like the tracking errors and ticket assignment. I have never used bugsnag but it looks like it does not have a supported rust platform on their site. Sentry does, however I think we are going to run into issues with trying to add sentry sdk into wingc because it has dependencies on openssl and from initial searching online it looks like thats not supported in our wasi target 

Could report the errors from the wing cli level but I think we might have lossy info 🤷

**Comment by eladb** [comment]
@MarkMcCulloh do you know what's the best practice for VSCode extensions to report crashes?

**Comment by MarkMcCulloh** [comment]
I also agree with Sentry being the best approach since it would work well for every part of the stack (cli, vscode, console, even winglang.io)

> @MarkMcCulloh do you know what's the best practice for VSCode extensions to report crashes?

Most extensions try _really_ _really_ hard to avoid crashes (many of them are written in js, so don't need to worry about panics). If they can't, you'll typically get a popup on the lower right-hand side asking if you want to report the crash. Manual reports like this can also be handled by sentry.
